src/charm.py

```diff
@@ -334,6 +395,16 @@ def render_manifest(self, manifest, name):
             yaml.safe_dump_all(manifest, f)
         return destination
 
+    def render_template(self, filename, **kwargs):
+
+        os.makedirs("templates/rendered", exist_ok=True)
+        destination = f"templates/rendered/{filename}_rendered.yaml"
+        template = self.jinja2_environment.get_template(filename)
+        content = template.render(**kwargs)
+        with open(destination, mode="w") as out:
+            out.write(content)
```

Comment:
I see you're writing to a file here.  I think I've seen this done like so:

```suggestion
        template.stream(**kwargs).dump(destination)
```